### ISSUE TITLE
Railway vs. Exceptions

### ISSUE THREAD

**Issue by dlazerka** [swlaschin/RailwayOrientedProgramming#3]
Hi Scott,

It's more of question than an issue.

I couldn't stop thinking that with propagating error results using railway, we're basically reinventing exceptions. With two main differences:
* We have to propagate exception up the stack manually.
* We lose the very handy feature of exceptions -- stacktraces. In large projects it's often hard to track where an error came from (e.g. "DB timeout" error due to slow query -- which query was that).
 
Can we get our stacktraces back with Railway? The only way I see is to have "cause" field at each exception, and wrap them errors manually before returning (with some custom differentiator aka "line number").

**Comment by swlaschin** [comment]
I agree with your points! In fact I wrote a whole post about when NOT to use Results :)
https://fsharpforfunandprofit.com/posts/against-railway-oriented-programming/

In short, use Result for domain modeling and use exceptions if you need diagnostics. Hope that helps!
